language/expressions.md

```diff
@@ -720,9 +726,26 @@ result. Chained assignments are permitted.
 
 Examples:
 
+~~~puppet
+
     $a = 10
     $x = $y = 0
 
+    # from array
+    [$a, $b] = [1, 2]  # assigns 1 to $a, and 2 to $b
+
+    # from hash
+    [$a, $b] = { a => 10, b => 20, c => 30 }  # assigns 10 to $a, and 20 to $b
+
+    # from class
+    class mymodule::someclass::example($x = 100) {
+      $a = 10
+    }
+    include example
+    [$a, $x] = Class['mymodule::someclass::example']  # assigns 10 to $a, and 100 to $x
```

Comment:
surprised me that this is fetching *all* variables in the class and not just parameters, kind of leaky about the internals of a class? 

Of course you could just fetch `$mymodule::someclass::example::a` so it's no worse than before.